src/Microsoft.TestPlatform.TestHostProvider/Hosting/DefaultTestHostManager.cs

```diff
@@ -176,6 +183,11 @@ public TestHostConnectionInfo GetTestHostConnectionInfo()
         /// <inheritdoc/>
         public IEnumerable<string> GetTestPlatformExtensions(IEnumerable<string> sources, IEnumerable<string> extensions)
         {
+            if (sources != null && sources.Any() && this.projectOutputExtensionsRequired)
```

Comment:
Does this work for not isolation scenario?

Reply:
Yes.